core/src/services/config.js

```diff
@@ -1,85 +1,96 @@
-import { isFunction, isPromise } from '../utilities/helpers.js';
+import { getConfigValueFromObjectAsync } from '../utilities/async-helpers';
+import { getConfigValueFromObject } from '../utilities/helpers';
 
-(function() {
-  window.Luigi = {
-    setConfig: configInput => {
-      window.Luigi.config = configInput;
-    },
-    getConfig: () => {
-      return window.Luigi.config;
-    }
-  };
-})();
-/*
- * Gets value of the given property on the given object.
- */
-export const getConfigValueFromObject = (object, property) => {
-  let propIndex = 0;
-  let nextValue = object;
-  const propertyPath = property.split('.');
-  while (nextValue && propIndex < propertyPath.length) {
-    nextValue = nextValue[propertyPath[propIndex++]];
+class LuigiConfigManager {
+  constructor() {
+    this.configReadyTimeout = {
+      valueMs: 2000,
+      id: undefined
+    };
+    this.config;
+
+    this.configReadyCallback = function() {};
   }
-  return nextValue;
-};
 
-/*
- * Gets value of the given property on Luigi config object.
- */
-export const getConfigValue = property => {
-  return getConfigValueFromObject(window.Luigi.config, property);
-};
+  setConfigCallbacks(configReadyCallback) {
+    this.configReadyCallback = configReadyCallback;
+    this.configReadyTimeout.id = setTimeout(() => {
+      // Avoid Luigi initialization if timeout reached
+      this.configReadyCallback = function() {};
+      this.configNotReadyCallback();
+    }, 2000);
+  }
 
-/*
- * Gets boolean value of the given property on Luigi config object.
- * Function return true if the property value is equal true or 'true'. Otherwise the function returns false.
- */
-export const getConfigBooleanValue = property => {
-  const configuredValue = getConfigValueFromObject(
-    window.Luigi.config,
-    property
-  );
-  if (configuredValue === true || configuredValue === 'true') {
-    return true;
+  setConfig(configInput) {
+    // console.log("Set config");
+    clearTimeout(this.configReadyTimeout.id);
+    this.config = configInput;
+    this.configReadyCallback();
   }
-  return false;
-};
 
-/*
- * Gets value of the given property on the given object.
- * If the value is a Function it is called and the result of that call is the value.
- * If the value is not a Promise it is wrapped to a Promise so that the returned value is definitely a Promise.
- */
-export const getConfigValueFromObjectAsync = (
-  object,
-  property,
-  ...parameters
-) => {
-  let value = getConfigValueFromObject(object, property);
-  if (isFunction(value)) {
-    value = value.apply(this, parameters);
-    if (isPromise(value)) {
-      return value;
+  getConfig() {
```

Comment:
You can be more _modern_ here and use a getter.
```
 get Config() {
    return this.config;
  }
```
and then
`this.getConfig()` -> `this.Config`.

I'm not requesting this change but I'd like you to know it :)

Reply:
I consider that option, but I see unnecessary to create a breaking change for this tiny improvement.